cmd/infra/azure/create.go

```diff
@@ -610,6 +615,53 @@ func createRhcosImages(ctx context.Context, l logr.Logger, o *CreateInfraOptions
 	return bootImageID, nil
 }
 
+// copyImageAndWait copies an RHCOS image from its Azure blob URL to a page blob within the managed resource group to be
+// used as the basis for creating Azure virtual machines for a NodePool.
+//
+// This function is hardcoded to wait 10 minutes for the copy to complete or else it will error out.
+func copyImageAndWait(ctx context.Context, rhcosURL string, pageBlobClient *pageblob.Client) error {
+	_, err := pageBlobClient.CopyFromURL(ctx, rhcosURL, nil)
+	if err != nil {
+		return fmt.Errorf("failed to start the process to copy rhcos image to vhd blob: %w", err)
+	}
+
+	timeout := 10 * time.Minute
+	startTime := time.Now()
+
+	for {
```

Comment:
should this be done with https://github.com/openshift/hypershift/blob/main/vendor/k8s.io/apimachinery/pkg/util/wait/poll.go#L45